core/src/test/java/org/elasticsearch/index/query/AbstractQueryTestCase.java

```diff
@@ -552,6 +552,12 @@ public void testToQuery() throws IOException {
         }
     }
 
+    private QueryBuilder<?> rewriteQuery(QB queryBuilder, QueryRewriteContext rewriteContext) throws IOException {
+        QueryBuilder<?> rewritten = QueryBuilder.rewriteQuery(queryBuilder, rewriteContext);
+        assertSerialization(rewritten);
```

Comment:
I see what it is doing, I was wondering why we are doing it given that every single query (that we might rewrite to) is already tested. More coverage? Potentially untested queries?